File: app/src/main/java/com/drawable/Square.java
package com.drawable;

import com.util.Color;

@SuppressWarnings({"unused", "WeakerAccess"})
public class Square extends Rectangle {
    public Square(double x, double y, int sideLength, int color) {
        this(x, y, sideLength, color, 0, 0);
    }

    public Square(double x, double y, int sideLength, Color color) {
        this(x, y, sideLength, color.getInt(), 0, 0);
    }

    public Square(double x, double y, int sideLength, int color, double xVelocity, double yVelocity) {
        super(x, y, sideLength, sideLength, color, xVelocity, yVelocity);
    }

    public Square(double x, double y, int sideLength, Color color, double xVelocity, double yVelocity) {
        this(x, y, sideLength, color.getInt(), xVelocity, yVelocity);
    }

    public Square(Square s) {
        super(s);
    }

    @Override
    public boolean equals(Object o) {
        if (!(o instanceof Square)) return false;
        Square other = (Square) o;
        return this.getColor() == other.getColor();
    }
}

File: app/src/main/java/com/drawable/Square.kt
package com.drawable

import com.util.Color

@Suppress("unused", "NAME_SHADOWING", "EqualsOrHashCode")
class Square : Rectangle {

    constructor(x: Double, y: Double, sideLength: Int, color: Color) : this(x, y, sideLength, color.int, 0.0, 0.0)

    @JvmOverloads constructor(x: Double, y: Double, sideLength: Int, color: Int, xVelocity: Double = 0.0, yVelocity: Double = 0.0) :
            super(x, y, sideLength, sideLength, color, xVelocity, yVelocity)

    constructor(x: Double, y: Double, sideLength: Int, color: Color, xVelocity: Double, yVelocity: Double) :
            this(x, y, sideLength, color.int, xVelocity, yVelocity)

    constructor(s: Square) : super(s)

    override fun equals(other: Any?): Boolean {
        if (other !is Square) return false
        val other = other as Square?
        return this.color == other!!.color
    }
}